docs/proposals/20221205-new-repo-koordetector.md

```diff
@@ -0,0 +1,136 @@
+
+OpenAnolis: https://gitee.com/anolis
+
+Prometheus: https://prometheus.io/
+
+## Summary
+
+To support Koordinator's Interference Detection feature, some advanced metrics need to be collected by multiple technologies such as CPU schedule latency by eBPF. This proposal is intended to develop a new sub-repo for Koodinator named `Koordetector` which covers many underlying kernel metrics collection tools implemented with eBPF technology or others, and expands Koordinator's ability to collect metrics in the form of plug-ins.
+
+## Motivation
+
+Enlighted by encapsulating the eBPF program of the CPU schedule latency collector into a separate tool, we noticed that Koordinator may better develop a new sub-repo to do this kind of jobs independently. On one hand,  given that the eBPF technology itself is in a stage of rapid development, and it has also attracted a lot of attention from kernel workers, Koordinator can use such sub-repo as a cutting edge for exploration in this regard. Furthermore, the sub-repo can be connected to mature third-party monitoring tools such as the `SysAK` project of the OpenAnolis community. Therefore, the proposal aims to decouple the ability of Koordinator to collect the underlying indicators of the kernel, and leave the professional work to professional people.
+
+When it comes to interference detection and solving, it should include several steps: metrics collection, metrics analysis and strategy execution. The new sub-repo can do more than just metrics collection, but also analysis. That's why we can name this repo as `Koordetector`. The special part of metrics collection in Koordetctor is that it provides **advanced metrics** like `CPU schedule latency`, `memory allocation latency`, `cache miss`, which are either related to underlying OS or need to be collected by advanced methods like eBPF. And there possibly is no need to do migration for current metrics collected by `Koodlet`. 
+
+The responsibilities of `Koordetector` and `Koordlet` are:
+
+- `Koordetector`: metrics collection and analysis, and draws a conclusion about whether some pods get interference
+- `Koordlet`: receive the result of interference detection from `Koordetector` committed strategy(resource suppress/evict)
+
+### Goals
+
+- Define the concepts, responsibilities and basic structure of the new repo `Koordetector`
+
+### Non-Goals/Future Work
+
+- [Future Plan] Design metric reporting CRDs or APIs. 
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a Koordinator developer, I want to commit my own metric collector implementation to enhance Koordlet's ability. 
+
+#### Story 2
+
+As a cluster administrator, I want to use Koordinator with small cost and control of the metric collectors. 
+
+### Design
+
+![koordetector](../images/koordetector.jpg)
+
+The whole structure of `Koordetector` looks similar to `Koordlet`, but without `QosManager` related components. `StateInformer`, `MetricsCache` are also necessary for collectors to run. 
```

Comment:
just introduce koordetector itself, no need to compare with others.
FYI:
`advisor manager`: manage a set of metrics collector, and also support external tools as prometheus format.
`tsdb`: cache data for analysis.
`analyser`: analysis metrics to detect interference.
`exporter`: expose data API for koordlet, and report to `koord metric server` in the future.